woocommerce-services.php

```diff
@@ -1040,7 +1040,8 @@ public function add_tracking_info_to_emails( $order, $sent_to_admin, $plain_text
 			// Generate a table row for each label
 			foreach ( $labels as $label ) {
 				$carrier = $label['carrier_id'];
-				$carrier_label = strtoupper( $carrier );
+				$carrier_service = $this->get_service_schemas_store()->get_service_schema_by_id( $carrier );
+				$carrier_label = ( ! empty( $carrier_service->carrier_name ) ) ? $carrier_service->carrier_name : strtoupper( $carrier );
```

Comment:
(minor) it could happen that `$carrier_service` is returned as `null`.
I think it would be safer if we checked for that case
```suggestion
				$carrier_label = ( ! $carrier_service || empty( $carrier_service->carrier_name ) ) ? strtoupper( $carrier ) : $carrier_service->carrier_name;
```

Reply:
Awesome, thanks for this! I had waffled a bit on this conditional, this was what I was looking for!

Reply:
Thanks for considering it! 👍